### PR TITLE
avb_streamhandler: Fix potential nullptr dereference

### PR DESCRIPTION
**Pull request by tvyas**
Signed-off-by: Tarun Vyas <tarun.vyas@intel.com>

### CODE REVIEW

**Review by keerockl — APPROVED** [review]

File: private/src/avb_streamhandler/IasAlsaHwDeviceHandler.cpp [hunk 1]
```diff
@@ -234,10 +234,17 @@ IasAvbProcessingResult IasAlsaHwDeviceHandler::init(uint16_t numChannels, uint32
...
-         mLastPtpEpoch     = mPtpProxy->getEpochCounter();
-         IasAvbStreamHandlerEnvironment::getConfigValue(IasRegKeys::cAudioTstampBuffer, mDescMode);
+         if (mPtpProxy)
+         {
+           mOptimalFillLevel = optimalFillLevel;
+           mAlsaDeviceType   = alsaDeviceType;
+           mLastPtpEpoch     = mPtpProxy->getEpochCounter();
+           IasAvbStreamHandlerEnvironment::getConfigValue(IasRegKeys::cAudioTstampBuffer, mDescMode);
+         }
+	 else
```
Comment (edersondisouza) [inline private/src/avb_streamhandler/IasAlsaHwDeviceHandler.cpp]: This else appears mis-indented.

File: private/src/avb_streamhandler/IasAlsaHwDeviceHandler.cpp [hunk 2]
```diff
@@ -234,10 +234,17 @@ IasAvbProcessingResult IasAlsaHwDeviceHandler::init(uint16_t numChannels, uint32
        if (eIasAvbProcOK == ret)
        {
          IasLibPtpDaemon* mPtpProxy = IasAvbStreamHandlerEnvironment::getPtpProxy();
-         mOptimalFillLevel = optimalFillLevel;
-         mAlsaDeviceType   = alsaDeviceType;
-         mLastPtpEpoch     = mPtpProxy->getEpochCounter();
-         IasAvbStreamHandlerEnvironment::getConfigValue(IasRegKeys::cAudioTstampBuffer, mDescMode);
+         if (mPtpProxy)
```
Comment (edersondisouza) [inline private/src/avb_streamhandler/IasAlsaHwDeviceHandler.cpp:237]: Minor nitpick: i think that in this project, they usually do the `x != nullptr` or even the `nullptr != x` (that I think it's ugly). But it's minor, you can merge without changing this.

**Review by edersondisouza — APPROVED** [review]

File: private/src/avb_streamhandler/IasAlsaHwDeviceHandler.cpp (hunk 1 above)
Comment (tvyas) [reply private/src/avb_streamhandler/IasAlsaHwDeviceHandler.cpp]: I took good care of making sure that it doesnt happen, but it slipped through the cracks  :(.

File: private/src/avb_streamhandler/IasAlsaHwDeviceHandler.cpp (hunk 2 above)
Comment (tvyas) [reply private/src/avb_streamhandler/IasAlsaHwDeviceHandler.cpp:237]: Right, the nullptr !=x hurt my eyes a lot, so I decided to completely do away with it.

File: private/src/avb_streamhandler/IasAlsaHwDeviceHandler.cpp (hunk 1 above)
Comment (tvyas) [reply private/src/avb_streamhandler/IasAlsaHwDeviceHandler.cpp]: Fixed ! Thanks for the review.

**Pull request merged**